src/Serilog.Sinks.ElmahIo/Sinks/ElmahIo/ElmahIOSink.cs

```diff
@@ -15,81 +15,97 @@
-                {
-                    Title = logEvent.RenderMessage(_formatProvider),
-                    Severity = LevelToSeverity(logEvent),
-                    DateTime = logEvent.Timestamp.DateTime.ToUniversalTime(),
-                    Detail = logEvent.Exception?.ToString(),
-                    Data = PropertiesToData(logEvent),
-                    Type = Type(logEvent),
-                    Hostname = Hostname(logEvent),
-                    Application = Application(logEvent),
-                    User = User(logEvent),
-                    Source = Source(logEvent),
-                    Method = Method(logEvent),
-                    Version = Version(logEvent),
-                    Url = Url(logEvent),
-                    StatusCode = StatusCode(logEvent),
-                };
+            if (events == null || !events.Any())
+                return;
 
-                _client.Messages.CreateAndNotify(_logId, message);
+            var client = _client;
+            if (_client == null)
+            {
+                ElmahioAPI api = new ElmahioAPI(new ApiKeyCredentials(_options.ApiKey), HttpClientHandlerFactory.GetHttpClientHandler());
+                api.HttpClient.Timeout = new TimeSpan(0, 0, 5);
+                api.HttpClient.DefaultRequestHeaders.UserAgent.Clear();
+                api.HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(new ProductHeaderValue("Serilog.Sinks.ElmahIo", _assemblyVersion)));
+                client = api;
             }
-            catch (Exception e)
+
+            foreach (var logEvent in events)
             {
-                Debugging.SelfLog.WriteLine("Caught exception while emitting to sink: {0}", e);
-                throw;
+                try
+                {
+                    var message = new CreateMessage
+                    {
+                        Title = logEvent.RenderMessage(_options.FormatProvider),
+                        Severity = LevelToSeverity(logEvent),
+                        DateTime = logEvent.Timestamp.DateTime.ToUniversalTime(),
+                        Detail = logEvent.Exception?.ToString(),
+                        Data = PropertiesToData(logEvent),
+                        Type = Type(logEvent),
+                        Hostname = Hostname(logEvent),
+                        Application = Application(logEvent),
+                        User = User(logEvent),
+                        Source = Source(logEvent),
+                        Method = Method(logEvent),
+                        Version = Version(logEvent),
+                        Url = Url(logEvent),
+                        StatusCode = StatusCode(logEvent),
+                    };
+
+                    client.Messages.CreateAndNotify(_options.LogId, message);
+                }
+                catch (Exception e)
+                {
+                    Debugging.SelfLog.WriteLine("Caught exception while emitting to sink: {0}", e);
+                    throw;
```

Comment:
Should you throw here?

Reply:
I'm not really sure. That is what the sink is doing today. But I don't know if errors from a sink should be thrown back to Serilog.

Reply:
Yeah I noticed that too. I see you removed it and I think that is the better approach. Logging should not break the app.